### PR TITLE
Enhance Spark JDBC examples

### PR DESCRIPTION
**Pull request by BentsiLeviav**
## Summary
This PR adds code examples in 4 different APIs (Java, Scala, Python, and SQL) to the Spark JDBC section.
In addition, a table of contents was added to the top of the doc, and a few heading mismatches were addressed.

### CODE REVIEW

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 1]
```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
...
-        df.show();
 
         // Stop the Spark session
         spark.stop();
     }
 ```
+</TabItem>
+<TabItem value="Scala" label="Scala">
+
+```java
```
Comment (mzitnik) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:487]: this should be scala

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 2]
```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
...
+        //---------------------------------------------------------------------------------------------------
+        // Load the table from ClickHouse using jdbc method
+        //---------------------------------------------------------------------------------------------------
         Properties jdbcProperties = new Properties();
         jdbcProperties.put("user", "default");
         jdbcProperties.put("password", "123456");
 
-        // Load the table from ClickHouse
-        Dataset<Row> df = spark.read().jdbc(jdbcUrl, "example_table", jdbcProperties);
+        Dataset<Row> df1 = spark.read().jdbc(jdbcURL, String.format("(%s)", query), jdbcProperties);
```
Comment (mzitnik) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:461]: I think we should write meaningful variables names

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 3]
```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
...
+
+        df1.show();
+
+        //---------------------------------------------------------------------------------------------------
+        // Load the table from ClickHouse using load method
+        //---------------------------------------------------------------------------------------------------
+        Dataset<Row> df2 = spark.read()
```
Comment (mzitnik) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:468]: I think we should write meaningful variables names

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 4]
```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
...
+
+
+  //---------------------------------------------------------------------------------------------------
+  // Load the table from ClickHouse using jdbc method
+  //---------------------------------------------------------------------------------------------------
+  val connectionProperties = new Properties()
+  connectionProperties.put("user", "default")
+  connectionProperties.put("password", "123456")
+
+  val df1: Dataset[Row] = spark.read.
```
Comment (mzitnik) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:503]: I think we should write meaningful variables names

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 5]
```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
...
+  //---------------------------------------------------------------------------------------------------
+  val df2: Dataset[Row] = spark.read
+    .format("jdbc")
+    .option("url", jdbcURL)
+    .option("user", "default")
+    .option("password", "123456")
+    .option("query", query)
+    .load()
+
+  df2.show()
```
Comment (mzitnik) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:518]: I think we should write meaningful variables names

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 6]
```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
...
+    .config("spark.jars", ",".join(jar_files)) \
+    .getOrCreate()
+
+url = "jdbc:ch://localhost:8123/default"
+user = "your_user" 
+password = "your_password"  
+query = "select * from example_table where id > 2"
+driver = "com.clickhouse.jdbc.ClickHouseDriver"
+
+df = (spark.read
```
Comment (mzitnik) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:553]: I think we should write meaningful variables names

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 7]
```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
...
         rows.add(RowFactory.create(2, "Doe"));
 
+
         Dataset<Row> df = spark.createDataFrame(rows, schema);
 
+        //---------------------------------------------------------------------------------------------------
+        // Write the df to ClickHouse using the jdbc method
+        //---------------------------------------------------------------------------------------------------
+
         df.write()
```
Comment (mzitnik) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:619]: I think we should write meaningful variables names

File: docs/en/integrations/data-ingestion/apache-spark/index.md [hunk 8]
```diff
@@ -426,68 +436,335 @@ One of the most used data sources supported by Spark is JDBC.
...
+
 
         // Stop the Spark session
         spark.stop();
     }
 ```
+</TabItem>
+<TabItem value="Scala" label="Scala">
+
+```java
```
Comment (mzitnik) [inline docs/en/integrations/data-ingestion/apache-spark/index.md:645]: this should be scala

**Review by mzitnik — COMMENTED** [review]
I would also ask to start clickHouse as perquisites

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 3 above)
Comment (BentsiLeviav) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:468]: Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 2 above)
Comment (BentsiLeviav) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:461]: Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 4 above)
Comment (BentsiLeviav) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:503]: Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 5 above)
Comment (BentsiLeviav) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:518]: Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 6 above)
Comment (BentsiLeviav) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:553]: Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 7 above)
Comment (BentsiLeviav) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:619]: Do you have any ideas for meaningful variable names? it is a standard practice to call a DataFrame df

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 8 above)
Comment (BentsiLeviav) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:645]: Unfortunately, there is no `scala` option, so I used the closest one with the closest highlighting.

File: docs/en/integrations/data-ingestion/apache-spark/index.md (hunk 1 above)
Comment (BentsiLeviav) [reply docs/en/integrations/data-ingestion/apache-spark/index.md:487]: There is no `scala` option so I used the closest one that highlights the syntax the best.

**Pull request merged**